htdocs/fourn/commande/dispatch.php

```diff
@@ -886,20 +933,24 @@
 
 							print '<tr class="oddeven" name="'.$type.$suffix.'">';
 							print '<td>';
-							print '<input name="fk_commandefourndet'.$suffix.'" type="hidden" value="'.$objp->rowid.'">';
-							print '<input name="product_batch'.$suffix.'" type="hidden" value="'.$objp->fk_product.'">';
+							print '<input name="fk_commandefourndet'.$suffix.'" type="hidden" value="'.$objp->rowid.'" />';
```

Comment:
Do not use xml ending syntax, even if html5 is able to accept it, using pure html is better

Reply:
For entry of product/serial/qty for reception (and shipment), a more powerful way to manage multiple serial input is planned.
If you take a look at the page to enter a reception, you will see a picto "split" to split a line into several lines so we are able from this screen to dispatch several lot, qty, warehouse target.
For mass input we must keep this philosophy. What is planned is:
Having a button "Mass input" that open a popup. This popup will have one textarea empty by default. We must be able to enter (by copy/paste for example from an excel sheet) a content that is
LOT1
LOT2
LOT3
or 
LOT1;QTY1
LOT2;QTY2
LOT3;QTY3
or 
LOT1;QTY1;WAREHOUSEID1
LOT2;QTY2;WAREHOUSEID1
LOT3;QTY3;WAREHOUSEID1

When clicking on "Close popup", it will automatically cal the js method addDispatchLine() for each line entered, the same way it is done when we click on the "split" picto and the entered value into line is filled by js into this new line.
We must proceed like this for several reason:
- We will be able to use the mass werehouse selection even on lot entered massively, keeping ability to use different warehouses.
- Mass entry will share the same interface than manual entry with all the powerfull of manual entrie.
- We will be able to fill the popup textarea field by using a barcode reader by scanning several barcode in a serial process (it will make LOT1, LOT2, LOT3 appearing with still the ability to modify, fix manually while keeping scanning)
- We will be able also to load a file into this popup so to make mass entry o serial from a file; a third way to enter data, but with always using the same interface.
- Such an interface can be used at a lot of other place.

For the page Stock transfer and Stock movement, there is currently no picto and no js function but we can image to show the picto just after the field "lot serial". But your first change to use a textarea instead of input is 

Also can you try to split this PR is smaller pieces (atomic PR), it is easier to discus on each feature on problem on some changes does not block the others.

Reply:
ok i will look at all this... and will also make smaller PRs.
Thanks

Reply:
This is the discussion we had with @altatof . We plan to do this version later. We will see that.

Reply:
LOT1;QTY1;WAREHOUSEID1
LOT2;QTY2;WAREHOUSEID2
LOT3;QTY3;WAREHOUSEIDX
Ou
SerialNumber;1;WAREHOUSEID1

Reply:
We will propose a Workflow with screenshots

Reply:
![image](https://user-images.githubusercontent.com/3288514/114156073-796c2380-9922-11eb-924c-62cedec5c1bd.png)

Reply:
I think we need more information like : 
WAREHOUSEIDX,Qty,Lot-orserialnumber;DMD;DLC;Prix;Projet;libellé;codemvt